core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -650,8 +662,12 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
                                       parentContext: RawlsRequestContext
   ): Future[Option[String]] =
     for {
-      _ <- assertNoChildrenBlockingWorkspaceDeletion(workspaceContext)
-
+      _ <-
+        if (!isAzureMcWorkspace(maybeMcWorkspace)) {
```

Comment:
It is questionable what we should do for McWorkspaces that are GCP, but we are already using this check further down in the method before calling `maybeDeleteGoogleProject`.

Reply:
Good thing to flag though I'm not too worried right now. I presume much of this functionality will be moved into wsm for gcp workspaces and wsm knows/will know which cloud platform a workspace is hosted on.

Reply:
More stylistic sugar (read: not important, but if you like) You can also use `Applicative` to handle these types of conditional effects. 
```scala
Applicative[Future].whenA(condition: Bool)(action: => Future[Unit])
Applicative[Future].unlessA(condition: Bool)(action: => Future[Unit])
```

Reply:
Thanks, much nicer.